proxy/http/HttpTransact.cc

```diff
@@ -5723,9 +5723,13 @@ HttpTransact::initialize_state_variables_from_request(State *s, HTTPHdr *obsolet
     s->cache_info.action = CACHE_DO_NO_ACTION;
   }
 
+  bool do_increment_stat = (0 == s->method);
+
   s->method = incoming_request->method_get_wksidx();
 
-  if (s->method == HTTP_WKSIDX_GET) {
+  if (!do_increment_stat) {
```

Comment:
Why adding test requires this change? Does this actually fix something?

Reply:
Yes , it fixes the associated issue.

Reply:
Name the PR accordingly. Who expects behavior change from the current title.